astra/src/main/java/com/slack/astra/graphApi/GraphConfig.java

```diff
@@ -141,23 +163,33 @@ public static GraphConfig load(String configYAML) throws IOException {
   }
 
   /**
-   * Creates metadata for a child node from a span, using either default behavior or configured
-   * mapping.
+   * Creates metadata for a graph entity (node or edge) from a span, using either default behavior
+   * or configured mapping.
    *
    * @param span ZipkinSpanResponse containing the span data
-   * @return SortedMap containing the metadata for the child node
+   * @return SortedMap containing the metadata for the requested entity
    */
-  public SortedMap<String, String> createMetadataFromSpan(ZipkinSpanResponse span) {
+  public SortedMap<String, String> createMetadataFromSpan(
+      ZipkinSpanResponse span, EntityType entityType) {
     SortedMap<String, String> metadata = new TreeMap<>();
 
     if (this == DEFAULT) {
-      // Default behavior: use service name from remote endpoint
-      metadata.put("service", span.getRemoteEndpoint().getServiceName());
+      // We only care about default behavior for a node, edge metadata is optional.
+      if (entityType == EntityType.NODE) {
+        // use service name from remote endpoint
+        metadata.put("service", span.getRemoteEndpoint().getServiceName());
+      }
     } else {
-      // Use configured tag mapping
       Map<String, String> tags = span.getTags();
-      for (String key : this.nodeMetadataTagMapping.keySet()) {
-        metadata.put(key, resolve(tags, key));
+
+      // Use configured tag mapping
+      Set<String> keys = this.nodeMetadataTagMapping.keySet();
+      if (entityType == EntityType.EDGE) {
+        keys = this.edgeMetadataTagMapping.keySet();
+      }
```

Comment:
You could use a switch expression for this
```suggestion
      Set<String> keys = switch(entityType) {
        case EDGE -> this.edgeMetadataTagMapping.keySet();
        case NODE -> this.nodeMetadataTagMapping.keySet();
      }
```